python/cudf/cudf/core/column/string.py

```diff
@@ -4506,6 +4506,30 @@ def __init__(
             children=children,
         )
 
+    def __sizeof__(self):
+        n = 0
+        if len(self.base_children) == 2:
+            child0_size = (self.size + 1) * self.base_children[
+                0
+            ].dtype.itemsize
+
+            if self.offset:
+                child1_size = (
+                    self.base_children[0][-1]
+                    - self.base_children[0][self.offset]
+                ) * self.base_children[1].dtype.itemsize
+            else:
+                child1_size = (
+                    self.base_children[0][-1] - self.base_children[0][0]
+                ) * self.base_children[1].dtype.itemsize
+
+            n += child0_size + child1_size
```

Comment:
We may want to do some profiling here as getting the 0th or last element in the offsets child requires a device --> host in this situation. We may need to memoize this if this function is called frequently. cc @quasiben

Reply:
By caching the property we seem to get a performance improvement:

For string column:
```python

s = cudf.Series(['hellow orld', 'ssdfsdf', 'sdfsd', 'sdfdsff', None] * 10000000)

This PR:

In [4]: %timeit s.memory_usage()
1.44 µs ± 9.96 ns per loop (mean ± std. dev. of 7 runs, 1000000 loops each)


branch-0.17:

In [8]: %timeit s.memory_usage()
2.41 µs ± 8.24 ns per loop (mean ± std. dev. of 7 runs, 100000 loops each)


```

For list column:

```python

s = cudf.Series([[[11, 22], [33, 44], [-10,20]], [[55, None, 11], [77, 88]]])



In [8]: s = cudf.Series([[[11, 22], [33, 44], [-10,20]], [[55, None, 11], [77, 88]]] * 10000000)

This PR: 

In [10]: %timeit s.memory_usage()
1.68 µs ± 17.8 ns per loop (mean ± std. dev. of 7 runs, 1000000 loops each)


branch-0.17:

In [3]: %timeit s.memory_usage()
3.4 µs ± 27.1 ns per loop (mean ± std. dev. of 7 runs, 100000 loops each)
```